admin/src/main/scala/io/buoyant/admin/PrometheusStatsHandler.scala

```diff
@@ -0,0 +1,52 @@
+package io.buoyant.admin
+
+import com.twitter.finagle.Service
+import com.twitter.finagle.http.{MediaType, Request, Response}
+import com.twitter.finagle.stats.MetricsStatsReceiver
+import com.twitter.io.Charsets
+import com.twitter.util.Future
+import org.jboss.netty.buffer.ChannelBuffers
+
+/**
+ * PrometheusStatsHandler
+ *
+ * Export Stats in a Prometheus-compatible format. Prometheus (http://prometheus.io/)
+ * exports stats in a somewhat different format than the Finagle's MetricsExporter. Specifically,
+ * instead of delimiting scopes by forward slashes, a colon is used. For stats, instead of exporting
+ * percentiles (p50, p99, etc.) and other quantities (max, min, etc.) as individual stats, they
+ * should be exported as labels in the Prometheus format.
+ */
+class PrometheusStatsHandler extends Service[Request, Response] {
+  private[this] lazy val registry = MetricsStatsReceiver.defaultRegistry
+  private[this] val StatPattern = """(.*)\.(.*?)$""".r
+  private[this] val ValidStatNames = Set("count", "sum", "avg", "min", "max", "stddev", "p50", "p90", "p95", "p99", "p9990")
+
+  import scala.collection.JavaConverters.mapAsScalaMapConverter
+
+  override def apply(request: Request): Future[Response] = {
+    val rsp = Response()
+    rsp.contentType = MediaType.Txt
+
+    val output = registry.sample().asScala.map {
+      case (key, value) => "%s %s".format(formatKey(key), value)
+    }.toList.sorted
+
+    rsp.contentString = output.mkString("\n") + '\n'
```

Comment:
Generally, It would be preferable to not have to iterate over this thing many times.

can we just build the response string in the formatting loop?